Update formal spec to (partially) implement the 3rd EH proposal (#180)
WebAssembly/exception-handling: document/core/syntax/instructions.rst

     \BLOCK~\blocktype~\instr^\ast~\END \\&&|&
     \LOOP~\blocktype~\instr^\ast~\END \\&&|&
     \IF~\blocktype~\instr^\ast~\ELSE~\instr^\ast~\END \\&&|&

>      \TRY~\blocktype~instr^\ast~(\CATCH~\tagidx~\instr^\ast)^\ast~(\CATCHALL~\instr^\ast)^?~\END \\ &&|&
>      \TRY~\blocktype~instr^\ast~\DELEGATE~\labelidx \\ &&|&

Reviewer: ```suggestion
     \TRY~\blocktype~\instr^\ast~(\CATCH~\tagidx~\instr^\ast)^\ast~(\CATCHALL~\instr^\ast)^?~\END \\ &&|&
     \TRY~\blocktype~\instr^\ast~\DELEGATE~\labelidx \\ &&|&
```

The |UNREACHABLE| instruction causes an unconditional :ref:`trap <trap>`.

The |BLOCK|, |LOOP|, |IF|, and |TRY| instructions are *structured* instructions.
They bracket nested sequences of instructions, called *blocks*,
separated by either |ELSE|, |CATCH|, or |CATCHALL| pseudo-instructions,
and terminated with either an |END| or a |DELEGATE| pseudo-instruction.
As the grammar prescribes, they must be well-nested.

The instructions |TRY|, |THROW|, and |RETHROW|, are concerned with handling exceptions.

> The |TRY| instructions create try-blocks, and either may handle or rethrow exceptions in the case of |CATCH| and |CATCHALL|,

Reviewer: ```suggestion
The |TRY| instructions install an exception handler, and may either handle exceptions in the case of |CATCH| and |CATCHALL|,
```

> or delegate exceptions to an outer potentially catching try-block in the case of |DELEGATE|.

Reviewer: ```suggestion
or rethrow them in an outer block in the case of |DELEGATE|.
```

> The |THROW| and |RETHROW| instructions alter control flow by searching for a matching handler in an enclosing catching-try block, if any.

Reviewer: ```suggestion
The |THROW| and |RETHROW| instructions alter control flow by searching for a matching handler in one of the enclosing |TRY| blocks, if any.
```
